diff --git a/Sources/FuzzilliCli/Profiles/KieselProfile.swift b/Sources/FuzzilliCli/Profiles/KieselProfile.swift
new file mode 100644
index 0000000..415482e
--- /dev/null
+++ b/Sources/FuzzilliCli/Profiles/KieselProfile.swift
@@ -0,0 +1,30 @@
+import Fuzzilli
+
+let kieselProfile = Profile(
+  processArgs: { randomize in return [""] },
+  processEnv: [:],
+  maxExecsBeforeRespawn: 1000,
+  timeout: Timeout.value(250),
+  codePrefix: """
+    """,
+  codeSuffix: """
+    """,
+  ecmaVersion: ECMAScriptVersion.es6,
+  startupTests: [
+    // Check that the fuzzilli integration is available.
+    ("fuzzilli('FUZZILLI_PRINT', 'test')", .shouldSucceed),
+
+    // Check that common crash types are detected.
+    ("fuzzilli('FUZZILLI_CRASH', 0)", .shouldCrash),
+    ("fuzzilli('FUZZILLI_CRASH', 1)", .shouldCrash),
+    ("fuzzilli('FUZZILLI_CRASH', 2)", .shouldCrash),
+  ],
+  additionalCodeGenerators: [],
+  additionalProgramTemplates: WeightedList<ProgramTemplate>([]),
+  disabledCodeGenerators: [],
+  disabledMutators: [],
+  additionalBuiltins: [:],
+  additionalObjectGroups: [],
+  additionalEnumerations: [],
+  optionalPostProcessor: nil
+)
diff --git a/Sources/FuzzilliCli/Profiles/Profile.swift b/Sources/FuzzilliCli/Profiles/Profile.swift
index 0947d95..1c1f6a9 100644
--- a/Sources/FuzzilliCli/Profiles/Profile.swift
+++ b/Sources/FuzzilliCli/Profiles/Profile.swift
@@ -54,4 +54,5 @@ let profiles = [
     "v8holefuzzing": v8HoleFuzzingProfile,
     "serenity": serenityProfile,
     "njs": njsProfile,
+    "kiesel": kieselProfile,
 ]
